service/controller/resource/capzcrs/create.go

```diff
@@ -238,8 +237,8 @@ func (r *Resource) mapAzureConfigToAzureMachine(ctx context.Context, cr provider
 
 func (r *Resource) updateCRs(ctx context.Context, crmappings []crmapping) error {
 	for _, m := range crmappings {
-		// Construct new instance by creating deep copy of desired object.
-		readCR := m.obj.DeepCopyObject()
+		// Construct new instance via reflection to ensure clean zero value object.
+		readCR := reflect.New(reflect.TypeOf(m.obj).Elem()).Interface().(runtime.Object)
```

Comment:
To decipher this:
- Get type of `m.obj`.
- Since we know it's a `runtime.Object`, it's guaranteed to be a pointer and we need to dereference it.
- Construct a new instance of it and get a pointer to that value (`New()`).
- Extract `interface{}` from `Value` object.
- Cast `interface{}` to `runtime.Object`.

Reply:
Maybe this explanation is worth a comment in the code.

Reply:
I rewrote it in steps and added comments.